app/uk/gov/hmrc/helptosave/connectors/DESConnector.scala

```diff
@@ -16,8 +16,71 @@
 
 package uk.gov.hmrc.helptosave.connectors
 
-import uk.gov.hmrc.http.HttpResponse
+import cats.Show
+import cats.syntax.show._
+import com.google.inject.{ImplementedBy, Inject, Singleton}
+import play.api.libs.json.{JsNull, JsValue, Writes}
+import uk.gov.hmrc.helptosave.config.{AppConfig, WSHttp}
+import uk.gov.hmrc.helptosave.metrics.Metrics
+import uk.gov.hmrc.helptosave.models.UCResponse
+import uk.gov.hmrc.helptosave.util.Time.nanosToPrettyString
+import uk.gov.hmrc.helptosave.util.{LogMessageTransformer, Logging, NINO, PagerDutyAlerting}
+import uk.gov.hmrc.http.{HeaderCarrier, HttpResponse}
 
+import scala.concurrent.{ExecutionContext, Future}
+
+@ImplementedBy(classOf[DESConnectorImpl])
 trait DESConnector {
   def desCorrelationId(response: HttpResponse): String = response.header("CorrelationId").getOrElse("-")
+
+  def isEligible(nino: String, ucResponse: Option[UCResponse])(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[HttpResponse]
+
+  def setFlag(nino: NINO)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[HttpResponse]
+
+  def getPersonalDetails(nino: NINO)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[HttpResponse]
+
+  def getThreshold()(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[HttpResponse]
+}
+
+@Singleton
+class DESConnectorImpl @Inject() (http:              WSHttp,
+                                  metrics:           Metrics,
+                                  pagerDutyAlerting: PagerDutyAlerting)(implicit transformer: LogMessageTransformer, appConfig: AppConfig)
```

Comment:
why do we still have metrics and pagerduty injected here ?

Reply:
same in HelpToSaveProxyConnector too

Reply:
we don't, sorry I overlooked these in the DesConnector. By removing them from the HelpToSaveProxyConnector, the metrics and pd logic would need to be moved out, I think that is beyond the scope of this ticket?

Reply:
ok, i am not sure why the ticket says inly about DES connectors, but i meant its about all connectors logic moving to services... better if we remove them from HelpToSaveProxyConnector too and move to services..